src/main/java/persistance/StorageManager.java

```diff
@@ -0,0 +1,218 @@
+package persistance;
+
+import org.hibernate.service.spi.ServiceException;
+
+import javax.persistence.*;
+import javax.persistence.EntityManager;
+import java.io.Serializable;
+
+public class StorageManager implements Serializable {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    public StorageManager() {
+
+    }
+
+    /**
+     * This function is for initialising the Entity Manager
+     * @return 0 for success, -1 for error
+     */
+    public boolean initEM() {
+        try {
+            EntityManagerFactory emf = Persistence.createEntityManagerFactory("fitbitPU");
+            em = emf.createEntityManager();
+        } catch (ServiceException se) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * This Function is for closing the Entity Manager.
+     */
+    public void closeEM() {
+        em.close();
+    }
+
+    /**
+     * This function will take a Token Map and either create one if the user doesn't already have a Token Map stored or
+     * update it.
+     * @param uID id of currently logged in user.
+     * @param accessToken access token to be stored.
+     * @param refreshToken refresh token to be stored.
+     */
+    public void commitTokenMap(String uID, String accessToken, String refreshToken) {
+        TokenMap tokenMap = doesTokenMapExist(uID);
+        if (tokenMap == null) {
+            createTokenMap(uID, accessToken, refreshToken);
+        } else {
+            updateTokenMap(tokenMap, accessToken, refreshToken);
+        }
+    }
+
+    /**
+     * Gets the Access Token from Database
+     * @param uId user id of logged in user
+     * @param refresh boolean value TRUE for refresh token & FALSE for access token
+     * @return Access Token for user
+     */
+    public String getToken(String uId, boolean refresh) {
+        TokenMap tokenMap = doesTokenMapExist(uId);
+        if (refresh) {
+            return tokenMap.getRefreshToken();
+        } else {
+            return tokenMap.getAccessToken();
+        }
+    }
+
+    /**
+     * This function removes a record from persistent storage.
+     * @param uId user id of the current user
+     * @return 0 for success, -1 failure (record doesn't exist)
+     */
+    public int removeTokenMap(String uId) {
```

Comment:
Why does this return an integer to determine failure / success? Can we not use a true/false bool here?

Reply:
done